lib/spack/spack/ci.py

```diff
@@ -1305,21 +1302,17 @@ def main_script_replacements(cmd):
             with open(copy_specs_file, "w") as fd:
                 fd.write(json.dumps(buildcache_copies))
 
-        sorted_output = {}
```

Comment:
Do we know why this was getting sorted, and do we have a good reason to remove it?

Reply:
I guess for debugging by making the YAML output vaguely diff-able? GitLab couldn't care less what order the keys are in so AFAICT there's no technical reason to sort the keys.

Reply:
I think it wasn't about diff-ability, but maybe just made it easier for me to scan through them.  I don't think it's important.

Reply:
However, IIRC @haampie asked at some point if they could be topo-sorted instead of lexicographically.  At the time I guessed it was about being able to more easily truncate the file to build less stuff.